src/ComunicaEngine.js

```diff
@@ -30,9 +31,10 @@ export default class ComunicaEngine {
     let bindings;
     const next = async () => {
       if (!bindings) {
-        // Determine the document to query from the subject
-        const document = this.getDocument(await this._subject);
-        const sources = [{ type: 'file', value: document }];
+        // Determine the document to query from the subject if there is no source
+        const sources = this._source ?
```

Comment:
and this then becomes `sources = this._sources.length > 0 ? this._sources : [{ type: 'file', value: this.getDocument(await this._subject) }]`